Src/Support/Google.Apis.Auth.Tests/OAuth2/OidcTokenTests.cs

```diff
@@ -0,0 +1,129 @@
+﻿/*
+Copyright 2020 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Auth.OAuth2.Responses;
+using Google.Apis.Logging;
+using Google.Apis.Tests.Mocks;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Apis.Auth.Tests.OAuth2
+{
+    public class OidcTokenTests
+    {
+        [Fact]
+        public async Task FetchesAccessToken()
+        {
+            MockClock clock = new MockClock { UtcNow = new DateTime(2020, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc) };
+
+            TokenRefreshManager refreshManager = null;
+            refreshManager = new TokenRefreshManager(RefreshTokenAsync, clock, new NullLogger());
+            OidcToken token = new OidcToken(refreshManager);
+
+            Task<bool> RefreshTokenAsync(CancellationToken cancellationToken)
```

Comment:
I tend to put local methods after the body of the method (so after Assert.Equal in this case). Don't much mind whether you follow that or not, particularly for tests, but thought I'd mention it.